src/base/net/downloadhandler.cpp

```diff
@@ -147,25 +147,22 @@ void Net::DownloadHandler::checkDownloadSize(qint64 bytesReceived, qint64 bytesT
 
 bool Net::DownloadHandler::saveToFile(const QByteArray &replyData, QString &filePath)
 {
-    QTemporaryFile *tmpfile = new QTemporaryFile(Utils::Fs::tempPath() + "XXXXXX");
-    if (!tmpfile->open()) {
-        delete tmpfile;
```

Comment:
At this point this had "autoremove" set to true and it would delete any leftover file created by the constructor.

Reply:
Yes, I suppose the new code is correct too: the temp file isn't created before `tmpfile.open()` successful call.